interfaces/builtin/serial_port.go

```diff
@@ -208,6 +209,24 @@ func (iface *serialPortInterface) AutoConnect(*snap.PlugInfo, *snap.SlotInfo) bo
 	return true
 }
 
+func (iface *serialPortInterface) HotplugDeviceDetected(di *hotplug.HotplugDeviceInfo) (*hotplug.ProposedSlot, error) {
+	if di.Subsystem() == "tty" && serialDeviceNodePattern.MatchString(di.DeviceName()) {
+		slot := hotplug.ProposedSlot{
```

Comment:
There are `tty` devices that may not be usb devices. Can you please add an extra check for `ID_BUS=usb`. We can extend this to non-usb serial devices but I would prefer to start small and focused where we can sure it works correctly.

Reply:
Good point, done.